### PR TITLE
Use flushdb on a test redis db

### PR DESCRIPTION
**Pull request by kbaum**
Been struggling with some of the tests sporadically failing. 

I like this approach for cleaning redis as we easily start with an empty clean each time but we avoid clearing out the data in the default redis database.

### CODE REVIEW

File: test/coverband/reporters/base_test.rb [hunk 1]
```diff
@@ -47,45 +46,4 @@ class ReportsBaseTest < Minitest::Test
...
     assert_equal expected, Coverband::Reporters::Base.send(:get_current_scov_data_imp, store, roots)[:merged]
   end
-
-  ###
-  # This test uses real world example data which helped uncover a bug
-  # The copied data doesn't format easily and isn't worth the effort to meet
-  # string style
-  # rubocop:disable all
-  ###
-  test '#get_current_scov_data_imp merges multiples of file data' do
```
Comment (kbaum) [inline test/coverband/reporters/base_test.rb]: This test was failing with missing mock expectations. I struggled with it for a while and thought maybe it wasn't worth maintaining. Let me know what you think.

Comment (danmayer) [reply test/coverband/reporters/base_test.rb]: I guess if this blocks the PR comment out and I will take a look and see if I can quickly fix or I will remove

Comment (kbaum) [reply test/coverband/reporters/base_test.rb]: Ok sounds good.

File: test/test_helper.rb [hunk 2]
```diff
@@ -30,20 +30,24 @@
 
 module Coverband
   module Test
+    TEST_DB = 2
+
+    def self.redis
```
Comment (kbaum) [inline test/test_helper.rb]: @danmayer - added a test redis instance that all of our tests can use. This way we all our in the same db and our database is being cleaned out properly.

**Pull request merged**